src/main.rs

```diff
@@ -464,10 +466,46 @@ enum TestOutcome {
 }
 
 impl Toolchain {
+    fn default_nightly() -> Option<Date<Utc>> {
```

Comment:
Can we add a doc comment on this? Something like

This returns the date of the default toolchain, if it is a nightly toolchain. Returns None if the installed toolchain is not a nightly toolchain.